pkg/render/applicationlayer/applicationlayer_test.go

```diff
@@ -410,4 +413,176 @@ var _ = Describe("Tigera Secure Application Layer rendering tests", func() {
 			Expect(dikastesVolMounts).To(ContainElement(expected))
 		}
 	})
+
+	It("should render with default l7 WAF configuration", func() {
+		expectedResources := []struct {
```

Comment:
Highly recommend moving away from this model of test.

Use `ExpectResource` instead: https://github.com/tigera/operator/blob/e5450ce876012b3e2c4392b2e36f3fe71750849d/pkg/render/common/test/testing.go#L93-L105

Note the comment on `ExpectResourceTypeAndObjectMetadata`: https://github.com/tigera/operator/blob/e5450ce876012b3e2c4392b2e36f3fe71750849d/pkg/render/common/test/testing.go#L141-L142

Reply:
I'd expect this to be a `[]client.Object` instead of a `[]struct`, in order for ExpectResource / ExpectResources to behave corretly.

Reply:
Part of the intention is that we can use the actual structs - `v1.ServiceAccount`, etc - instead of "fake" structs that are more error prone.